catalog/spatial/wps/spatial-wps-endpoint/src/test/java/org/codice/ddf/spatial/ogc/wps/process/endpoint/WpsEndpointTest.java

```diff
@@ -0,0 +1,592 @@
+        extraData.setSchema("test-schema.txt");
+        extraDataInputType.setData(data);
+        extraDataInputType.setId("extrainput");
+        executeRequestType.getInput()
+                .add(dataInputType);
+        executeRequestType.getInput()
+                .add(extraDataInputType);
+        OutputDefinitionType dataOutputType = new OutputDefinitionType();
+        dataOutputType.setId(complexDataDesc.getId());
+        dataOutputType.setEncoding("UTF-8");
+        dataOutputType.setMimeType("text/plain");
+        dataOutputType.setSchema("test-schema.txt");
+        dataOutputType.setTransmission(DataTransmissionModeType.REFERENCE);
+        executeRequestType.setOutput(Collections.singletonList(dataOutputType));
+
+        Response response = wpsEndpoint.execute(executeRequestType);
+        verify(simpleProcess).asyncExecute(any(ExecutionRequest.class));
+        assertThat(response.getEntity(), is(not(nullValue())));
+    }
+
+    @Test
+    public void complexSyncExecuteUsingDefaults() throws Exception {
+        when(processRepository.getProcess(anyString())).thenReturn(complexProcess);
+        ExecuteRequestType executeRequestType = new ExecuteRequestType();
+        CodeType ct = new CodeType();
+        ct.setValue("complex");
+        executeRequestType.setIdentifier(ct);
+        executeRequestType.setResponse("document");
+        executeRequestType.setMode("sync");
+        DataInputType literalInputType = new DataInputType();
+        Data inLiteral = new Data();
+        literalInputType.setData(inLiteral);
+        literalInputType.setId("inLiteral");
+        Data inComplex = new Data();
+        DataInputType complexInputType = new DataInputType();
+        String inComplexData = "{key:\"value\"}";
+        inComplex.setContent(Collections.singletonList(inComplexData));
+        complexInputType.setData(inComplex);
+        complexInputType.setId("inComplex");
+        DataInputType groupInputType = new DataInputType();
+        groupInputType.setId("inGroup");
+        groupInputType.setInput(Arrays.asList(complexInputType, literalInputType));
+        executeRequestType.setInput(Arrays.asList(groupInputType, groupInputType, groupInputType));
+
+        Response response = wpsEndpoint.execute(executeRequestType);
+        ExecutionRequest executionRequest = complexProcess.executionRequest;
+        DataGroup dataGroup = (DataGroup) executionRequest.getInputData()
+                .get(0);
+
+        assertThat(dataGroup.getData(), Matchers.iterableWithSize(2));
+        //input order should be preserved
+        org.codice.ddf.spatial.process.api.request.Data complexArg = dataGroup.getData()
+                .get(0);
+        org.codice.ddf.spatial.process.api.request.Data literalArg = dataGroup.getData()
+                .get(1);
+        assertThat(literalArg, is(notNullValue()));
+        assertThat(complexArg, is(notNullValue()));
+        assertThat(complexProcess.inLiteral.getDataFormats()
```

Comment:
✏️ Could use `hasItem()` for better readability

Reply:
Doing something slightly weird comparing a child to its parent so hasItem doesn't work.